packages/backend/src/managers/llama-stack/llamaStackManager.spec.ts

```diff
@@ -0,0 +1,181 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import { TaskRegistry } from '../../registries/TaskRegistry';
+import { beforeAll, beforeEach, expect, test, vi } from 'vitest';
+import type { ContainerCreateResult, ContainerInfo, ImageInfo, TelemetryLogger } from '@podman-desktop/api';
+import { containerEngine, EventEmitter } from '@podman-desktop/api';
+import type { PodmanConnection } from '../podmanConnection';
+import { ContainerRegistry } from '../../registries/ContainerRegistry';
+import { TestEventEmitter } from '../../tests/utils';
+import { VMType } from '@shared/models/IPodman';
+import type { Task } from '@shared/models/ITask';
+import llama_stack_images from '../../assets/llama-stack-images.json';
+import type { RpcExtension } from '@shared/messages/MessageProxy';
+import { LLAMA_STACK_API_PORT_LABEL, LLAMA_STACK_CONTAINER_LABEL, LlamaStackManager } from './llamaStackManager';
+import { LLAMA_STACK_CONTAINER_TRACKINGID } from '@shared/models/llama-stack/LlamaStackContainerInfo';
+import type { ConfigurationRegistry } from '../../registries/ConfigurationRegistry';
+import type { ExtensionConfiguration } from '@shared/models/IExtensionConfiguration';
+
+vi.mock('@podman-desktop/api', () => {
+  return {
+    EventEmitter: vi.fn(),
+    containerEngine: {
+      listContainers: vi.fn(),
+      listImages: vi.fn(),
+      createContainer: vi.fn(),
+      onEvent: vi.fn(),
+    },
+  };
+});
+
+const taskRegistry = new TaskRegistry({ fire: vi.fn().mockResolvedValue(true) } as unknown as RpcExtension);
+
+const podmanConnection: PodmanConnection = {
+  onPodmanConnectionEvent: vi.fn(),
+  findRunningContainerProviderConnection: vi.fn(),
+} as unknown as PodmanConnection;
+
+const configurationRegistry = {
+  getExtensionConfiguration: vi.fn(),
+} as unknown as ConfigurationRegistry;
+
+const telemetryMock = {
+  logUsage: vi.fn(),
+  logError: vi.fn(),
+} as unknown as TelemetryLogger;
+
+let llamaStackManager: LlamaStackManager;
+
+beforeAll(() => {
+  vi.mocked(EventEmitter).mockImplementation(() => new TestEventEmitter() as unknown as EventEmitter<unknown>);
+});
+
+beforeEach(() => {
+  const containerRegistry = new ContainerRegistry();
+  containerRegistry.init();
+  llamaStackManager = new LlamaStackManager(
+    '',
+    taskRegistry,
+    podmanConnection,
+    containerRegistry,
+    configurationRegistry,
+    telemetryMock,
+  );
+  llamaStackManager.init();
+  taskRegistry.deleteByLabels({ trackingId: LLAMA_STACK_CONTAINER_TRACKINGID });
```

Comment:
I would rather using `taskRegistry.dispose()` as here we could have side effect between tests (made a PR https://github.com/containers/podman-desktop-extension-ai-lab/pull/2838)

I would even say we should probably re-create both the taskRegistry and the llamaStackManager in the beforeEach.

I would not include the init in the beforeEach, as you don't test that the init is doing what we expect (registering events listener)